src/Microsoft.DotNet.MSBuildSdkResolver/SemanticVersion.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    internal sealed class SemanticVersion
+    {
+        public int Major { get; private set; }
+        public int Minor { get; private set; }
+        public int Patch { get; private set; }
+        public string Pre { get; private set; }
+        public string Build { get; private set; }
```

Comment:
These are all set in constructor so you don't need `private set`. Makes it clearer that the object is immutable.